.readthedocs.yaml

```diff
@@ -14,7 +14,7 @@ build:
     pre_install:
       - git update-index --assume-unchanged docs/rtd_environment.yaml docs/conf.py
     post_install:
-      - towncrier build --keep
+      - git describe --exact-match || towncrier build --keep
```

Comment:
Thanks! It's an attempt to fix https://github.com/asdf-format/asdf/issues/1844 where `towncrier build --keep` fails to build for a tagged commit.
For the 4.0.0 release the addition of the `4.0.0` tag will trigger readthedocs to build docs for that version (similar to https://asdf.readthedocs.io/en/3.5.0/). With the last release (the 3.5.0) that one failed because the `3.5.0` tag contained a changelog section for 3.5.0 already and towncrier saw the current commit was tagged so it failed when it tried to add a second 3.5.0 section.
With this command `git describe --exact-match || towncrier build --keep` the hope is that readthedocs when building the 4.0.0 docs will run the command and since `git describe --exact-match` will succeed (since the commit is an exact match for a tag) the second part of the command won't be run. This means `towncrier build --keep` won't run for a tagged commit but I think that's fine since tagged commits are either:
- a commit for a release which will already have a changelog section for that version
- a tag added to main to trigger scm_version to use a newer version where temporarily (for only that tagged commit) the 'latest' docs won't contain the development changelog (but the next commit added to main after the tag will trigger a 'latest' with the changelog)